pkg/cli/internal/frameworks/typescript/templates/src/server.ts.tmpl

```diff
@@ -0,0 +1,91 @@
+import { Server } from '@modelcontextprotocol/sdk/server/index.js';
+import { StdioServerTransport } from '@modelcontextprotocol/sdk/server/stdio.js';
+import { CallToolRequestSchema, ListToolsRequestSchema } from '@modelcontextprotocol/sdk/types.js';
+import { zodToJsonSchema } from 'zod-to-json-schema';
+import { allTools } from './tools.js';
+
+export async function createServer(): Promise<Server> {
+  const server = new Server(
+    {
+      name: '{{.ProjectName}}',
+      version: '{{.Version}}',
+    },
+    {
+      capabilities: {
+        tools: {},
+      },
+    }
+  );
+
+  // List tools
+  server.setRequestHandler(ListToolsRequestSchema, async () => {
+    return {
+      tools: allTools.map(tool => ({
+        name: tool.name,
+        description: tool.description,
+        inputSchema: zodToJsonSchema(tool.inputSchema, { strictUnions: true }) as any,
+      })),
+    };
+  });
+
+  // Call tool
+  server.setRequestHandler(CallToolRequestSchema, async (request) => {
+    const { name, arguments: args } = request.params;
+    const tool = allTools.find(t => t.name === name);
+    if (!tool) {
+      return {
+        content: [{ type: 'text', text: `Tool ${name} not found` }],
+        isError: true,
+      };
+    }
+
+    try {
+      const result = await tool.handler(args || {});
```

Comment:
Building the docker image failed for me on this line with the error `Property 'message' is missing in type 'Record<string, unknown>' but required in type '{ message: string; }'.` I think we could probably just do the following instead 

```
const result = await tool.handler((args || {}) as any);
```